src/main/kotlin/app/revanced/patches/ad/VideoAdsPatch.kt

```diff
@@ -0,0 +1,103 @@
+package app.revanced.patches.ad
+
+import app.revanced.patcher.PatcherData
+import app.revanced.patcher.extensions.AccessFlagExtensions.Companion.or
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.patch.*
+import app.revanced.patcher.signature.MethodMetadata
+import app.revanced.patcher.signature.MethodSignature
+import app.revanced.patcher.signature.MethodSignatureMetadata
+import app.revanced.patcher.signature.PatternScanMethod
+import app.revanced.patcher.smali.asInstructions
+import org.jf.dexlib2.AccessFlags
+import org.jf.dexlib2.Opcode
+
+private val compatiblePackages = arrayOf("com.google.android.youtube")
+
+class VideoAdsPatch : Patch(
+    metadata = PatchMetadata(
+        shortName = "video-ads",
+        name = "YouTube Video Ads Patch",
+        description = "Patch to remove ads in the YouTube video player.",
+        compatiblePackages = compatiblePackages,
+        version = "0.0.1"
+    ),
+    signatures = listOf(
+        MethodSignature(
+            methodSignatureMetadata = MethodSignatureMetadata(
+                name = "show-video-ads-constructor",
+                methodMetadata = MethodMetadata(
+                    definingClass = "zai",
+                    name = "<init>",
+                ),
+                patternScanMethod = PatternScanMethod.Fuzzy(2),// FIXME: Test this threshold and find the best value.
+                compatiblePackages = compatiblePackages,
```

Comment:
What is this used for? It doesn't seem to have a use right now.

Reply:
The `compatible packages` parameter will be useful to filter for patches. There might be global patches, which edit a package's icon or resources, or generic patches Microg, which can be applied for multiple packages like YouTube Music or YouTube. A patch might also be compatible with one package only. The manager can use this metadata, to apply correct patches for each package accordingly.